service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/WorkspaceCloneUtils.java

```diff
@@ -51,19 +55,22 @@ public static WsmCloneResourceResult flightStatusToCloneResult(
    */
   public static ControlledAzureStorageContainerResource buildDestinationControlledAzureContainer(
       ControlledAzureStorageContainerResource sourceContainer,
-      UUID storageAccountId,
+      DestinationStorageAccount storageAccountInfo,
       UUID destinationWorkspaceId,
       UUID destinationResourceId,
       String name,
       @Nullable String description,
       String cloudInstanceName) {
-    return ControlledAzureStorageContainerResource.builder()
-        .storageContainerName(cloudInstanceName)
-        .storageAccountId(storageAccountId)
-        .common(
-            sourceContainer.buildControlledCloneResourceCommonFields(
-                destinationWorkspaceId, destinationResourceId, null, name, description))
-        .build();
+    var builder =
+        ControlledAzureStorageContainerResource.builder()
+            .storageContainerName(cloudInstanceName)
+            .common(
+                sourceContainer.buildControlledCloneResourceCommonFields(
+                    destinationWorkspaceId, destinationResourceId, null, name, description));
+    if (StorageAccountType.WORKSPACE == storageAccountInfo.storageAccountType()) {
```

Comment:
This is why we need the either semantic; we omit the storage account ID if we're creating the destination container in the landing zone.